File: Fam.ly - iOS Edition/Fam.ly - iOS Edition/Objects/Permission/Request.swift
//
//  Request.swift
//  Fam.ly - iOS Edition
//
//  Created by Johnny Mack on 6/12/19.
//  Copyright © 2019 John Mack. All rights reserved.
//

import Foundation

public class Request: Codable {
    
    
    
    // Class properties
    var requesterName: String
    var requesterId: String
    var requestMessage: String
    var requestStatus: Int
    var timeStamp: Date
    var firstApprover: String?
    
    
    
    // Constructor
    init(RequesterName requesterName: String, RequesterId requesterId: String, RequestMessage requestMessage: String, TimeStamp timeStamp: Date) {
        self.requesterName = requesterName
        self.requesterId = requesterId
        self.requestMessage = requestMessage
        self.requestStatus = 0
        self.timeStamp = timeStamp
    }
    
    
    
    // Getters / Setters
    public func getRequesterName() -> String {
        return requesterName
    }
    
    public func setRequesterName(RequesterName requesterName: String) {
        self.requesterName = requesterName
    }
    
    public func getRequesterId() -> String {
        return requesterId
    }
    
    public func setRequesterId(RequesterId requesterId: String) {
        self.requesterId = requesterId
    }
    
    public func getRequestMessage() -> String {
        return requestMessage
    }
    
    public func setRequestMessage(RequestMessage requestMessage: String) {
        self.requestMessage = requestMessage
    }
    
    public func getRequestStatus() -> Int {
        return requestStatus
    }
    
    public func setRequestStatus(RequestStatus requestStatus: Int) {
        self.requestStatus = requestStatus
    }
    
    public func getTimeStamp() -> Date {
        return timeStamp
    }
    
    public func setTimeStamp(TimeStamp timeStamp: Date) {
        self.timeStamp = timeStamp
    }
    
    public func getFirstApprover() -> String? {
        return firstApprover
    }
    
    public func setFirstApprover(FirstApprover firstApprover: String?) {
        self.firstApprover = firstApprover
    }
    
    
    
    // Custom methods
    public func getRequestId() -> String {
        return requesterId + timeStamp.toMillis()!.description
    }
}

File: FamlyAndroidEdition/app/src/main/java/com/mack/john/famly_androidedition/objects/permission_request/Request.java
package com.mack.john.famly_androidedition.objects.permission_request;

import com.mack.john.famly_androidedition.objects.post.Post;

import java.io.Serializable;
import java.util.Date;

public class Request implements Serializable, Comparable<Request> {



    // Class properties
    String requesterName;
    String requesterId;
    String requestMessage;
    int requestStatus; // 0 == Pending; 1 == Approved; 2 == Denied
    Date timeStamp;
    String firstApprover;



    // Constructor
    public Request(String requesterName, String requesterId, String requestMessage, Date timeStamp) {
        this.requesterName = requesterName;
        this.requesterId = requesterId;
        this.requestMessage = requestMessage;
        this.requestStatus = requestStatus = 0; // 0 == Pending; 1 == Approved; 2 == Denied
        this.timeStamp = timeStamp;
        this.firstApprover = null;
    }



    // System generated methods
    @Override
    public int compareTo(Request request) {
        return getTimeStamp().compareTo(request.getTimeStamp());
    }




    // Getters / Setters
    public String getRequesterName() {
        return requesterName;
    }

    public void setRequesterName(String requesterName) {
        this.requesterName = requesterName;
    }

    public String getRequesterId() {
        return requesterId;
    }

    public void setRequesterId(String requesterId) {
        this.requesterId = requesterId;
    }

    public String getRequestMessage() {
        return requestMessage;
    }

    public void setRequestMessage(String requestMessage) {
        this.requestMessage = requestMessage;
    }

    public int getRequestStatus() {
        return requestStatus;
    }

    public void setRequestStatus(int requestStatus) {
        this.requestStatus = requestStatus;
    }

    public Date getTimeStamp() {
        return timeStamp;
    }

    public void setTimeStamp(Date timeStamp) {
        this.timeStamp = timeStamp;
    }

    public String getFirstApprover() {
        return firstApprover;
    }

    public void setFirstApprover(String firstApprover) {
        this.firstApprover = firstApprover;
    }




    // Custom method
    public String getRequestId() {
        String requestId = getRequesterId() + getTimeStamp().getTime();

        return requestId;
    }
}
